File: GroceryShoppingListApp/src/GroceryShoppingListApp.Core/Model/PriceHistoryAggregate/PriceHistory.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace GroceryShoppingListApp.Core;

public class PriceHistory
{
    public Guid PriceHistoryId { get; set; }
    public Guid GroceryItemId { get; set; }
    public Guid StoreId { get; set; }
    public decimal Price { get; set; }
    public DateTime Date { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: GroceryShoppingListApp/src/GroceryShoppingListApp.WebApp/projects/grocery-shopping-list-app/src/app/models/price-history.ts
export interface PriceHistory {
  priceHistoryId: string;
  groceryItemId: string;
  storeId: string;
  price: number;
  date: Date;
  createdAt: Date;
}

export interface CreatePriceHistoryRequest {
  groceryItemId: string;
  storeId: string;
  price: number;
  date: Date;
}

export interface UpdatePriceHistoryRequest {
  priceHistoryId: string;
  groceryItemId: string;
  storeId: string;
  price: number;
  date: Date;
}
